### PR TITLE
WOADD-564 : Indication for resit

### PR DESCRIPTION
**Pull request by natividadelijah**
*** PLEASE DO NOT OPEN PULL REQUESTS VIA GITHUB ***

The moodle.git repository at Github is just a mirror of the official repository. We do not accept pull requests at Github.

See CONTRIBUTING.txt guidelines for how to contribute patches for Moodle. Thank you.

--

### CODE REVIEW

File: local/gugcat/classes/grade_aggregation.php [hunk 1]
```diff
@@ -136,4 +138,21 @@ public static function get_rows($course, $modules, $students){
...
+        $grade_->usermodified = $studentno;
+        $grade_->itemid = $aggradeid;
+        $grade_->userid = $studentno;
+        $grade_->timemodified = time();
+        if(is_null($grade_->information)){
+            $grade_->information = '1'; 
+        }  
+        else {
+            $grade_->information = null;
+        }
```
Comment (m-aquino) [inline local/gugcat/classes/grade_aggregation.php]: `$grade_->information = is_null($grade_->information) ? null : '1';`
One liner check would suffice.

File: local/gugcat/classes/grade_aggregation.php [hunk 2]
```diff
@@ -136,4 +138,21 @@ public static function get_rows($course, $modules, $students){
         }
         return $rows;
     }
+
+    public static function require_resit($studentno){
+        global $aggradeid;
+
+        $grade_ = new grade_grade(array('userid' => $studentno, 'itemid' => $aggradeid), true);
+        $grade_->usermodified = $studentno;
```
Comment (m-aquino) [inline local/gugcat/classes/grade_aggregation.php]: The usermodified is the the userid of the person who modified this grade, if so the current user.

File: local/gugcat/overview/index.php [hunk 3]
```diff
@@ -48,9 +48,21 @@
 $coursecontext = context_course::instance($course->id);
 $students = get_enrolled_users($coursecontext, 'moodle/competency:coursecompetencygradable');
 $activities = local_gugcat::get_activities($courseid);
-
 $rows = grade_aggregation::get_rows($course, $activities, $students);
 
+if(!empty($_POST)){
+    if(isset($_POST['resit']) && !empty($_POST['rowstudentno'])){
+        $studentno = $_POST['rowstudentno'];
+        $status = grade_aggregation::require_resit($studentno);
```
Comment (m-aquino) [inline local/gugcat/overview/index.php]: No need to assign it to another variable if it will not be used.

**Review by m-aquino — CHANGES_REQUESTED** [review]

**Review by m-aquino — APPROVED** [review]

File: local/gugcat/overview/index.php [hunk 4]
```diff
@@ -48,9 +48,20 @@
...
 $students = get_enrolled_users($coursecontext, 'moodle/competency:coursecompetencygradable');
 $activities = local_gugcat::get_activities($courseid);
-
 $rows = grade_aggregation::get_rows($course, $activities, $students);
 
+if(!empty($_POST)){
+    if(isset($_POST['resit']) && !empty($_POST['rowstudentno'])){
+        grade_aggregation::require_resit($_POST['rowstudentno']);
+        unset($_POST);
+        header("Location: ".$_SERVER['REQUEST_URI']);
```
Comment (lee-acn) [inline local/gugcat/overview/index.php:57]: Please try using redirect($url).

**Review by lee-acn — APPROVED** [review]

**Pull request merged**